helidon-cli/impl/src/main/java/io/helidon/build/cli/impl/DevCommand.java

```diff
@@ -13,52 +13,64 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.helidon.build.cli.impl;
 
-import java.util.Properties;
+package io.helidon.build.cli.impl;
 
 import io.helidon.build.cli.harness.Command;
 import io.helidon.build.cli.harness.CommandContext;
 import io.helidon.build.cli.harness.CommandExecution;
 import io.helidon.build.cli.harness.Creator;
 import io.helidon.build.cli.harness.Option.Flag;
+import io.helidon.build.util.ProjectConfig;
 
 /**
  * The {@code dev} command.
  */
 @Command(name = "dev", description = "Continuous application development")
 public final class DevCommand extends BaseCommand implements CommandExecution {
 
-    private static final String MAVEN_PLUGIN = "maven.plugin";
+    private static final String CLEAN_PROP_PREFIX = "-Ddev.clean=";
+    private static final String FORK_PROP_PREFIX = "-Ddev.fork=";
+    private static final String MAVEN_PLUGIN_PROPERTY = "maven.plugin";
+    private static final String DEV_GOAL = "dev";
+    private static final String DEFAULT_DEV_GOAL = "helidon:" + DEV_GOAL;
 
     private final CommonOptions commonOptions;
     private final boolean clean;
     private final boolean fork;
 
     @Creator
     DevCommand(
-            CommonOptions commonOptions,
-            @Flag(name = "clean", description = "Perform a clean before the first build") boolean clean,
-            @Flag(name = "fork", description = "Fork mvn execution") boolean fork) {
+        CommonOptions commonOptions,
+        @Flag(name = "clean", description = "Perform a clean before the first build") boolean clean,
+        @Flag(name = "fork", description = "Fork mvn execution") boolean fork) {
         this.commonOptions = commonOptions;
         this.clean = clean;
         this.fork = fork;
     }
 
     @Override
     public void execute(CommandContext context) {
-        Properties cliConfig = cliConfig();
-        String cleanProp = "-Ddev.clean=" + clean;
-        String forkProp = "-Ddev.fork=" + fork;
-        StringBuilder pluginGoal = new StringBuilder(cliConfig.getProperty(MAVEN_PLUGIN));
-        pluginGoal.append(":")
-                .append(helidonVersion())
-                .append(":dev");
+        final ProjectConfig projectConfig = projectConfig(commonOptions.project().toPath());
+        if (!projectConfig.exists()) {
+            context.exitAction(CommandContext.ExitStatus.FAILURE, "Unable to find project");
+            return;
+        }
+
+        final String cleanProp = CLEAN_PROP_PREFIX + clean;
+        final String forkProp = FORK_PROP_PREFIX + fork;
+        final String helidonVersion = projectConfig.property(HELIDON_VERSION);
```

Comment:
We should not need to specify the groupId and version.

The requirement may just be that the `helidon-maven-plugin` must be configured in the maven project so that `helidon:goal` just works, this way the version of the plugin is 100% derived from the project.

If we had configuration incompatibilities between plugin versions, doing it this way would ensure that we don't force a different version than the one configured in the user's project.

Since our current design a "glorified alias" we should make sure that the contract between the cli and plugin is never broken, (i.e the various properties we pass on the forked mvn invocation) ; maybe grouping these into a special class to emphasize that would be a good idea.

Reply:
My first pass at this just used `helidon:dev`, but that mapped to the latest version (and failed our test cases). The fix you propose means that `init` must currently add the `version.helidon.plugin` property to the application pom. I'll see how hard that is to do.

Reply:
@romain-grecourt Is your comment about the dev command or the init command? I don't understand the context here.

Reply:
This is about forking maven invocations to the helidon maven plugin without specifying the version. It applies to all commands except init.

Reply:
Adding the `version.helidon.plugin` property at `init` for now and in the template later seems like the right way to solve this.

Reply:
sounds good.